### PR TITLE
DO NOT MERGE - Simplify updating case a bit

### PR DESCRIPTION
**Pull request by rkondratowicz**
Proposing simplification to an open PR

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java [hunk 1]
```diff
@@ -301,16 +300,24 @@ private void attachCaseByCcdId(
...
-                ccdCaseUpdater.updateCase(
-                    exceptionRecord,
-                    serviceConfigItem,
-                    ignoreWarnings,
-                    idamToken,
-                    userId,
-                    targetCaseCcdRef
-                );
+                Validation<String, String> serviceNameValidationResult =
+                    hasServiceNameInCaseTypeId(exceptionRecordData);
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java]: There's no point validating service name after it's already been validated at an earlier stage of request processing.

Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java]: But still, this is an improvement to the original version

Comment (apereverzin) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java]: Leszek, are you not off today?

Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java]: This is clearer, done using less code and is some step towards removing validation (and handling its failure) from the place in the code that shouldn't be validating things which were already validated.

Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java]: This is clearer, done using less code and is some step towards removing validation (and handling its failure) from the place in the code that shouldn't be validating things which were already validated.

Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java]: Removing or not - extra if/else is not necessary. will certainly make even more clearance if used `.getOrElseThrow` right here. Just a third party suggestion here, not a blocker comment

Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java]: There will be nothing to invoke `getOrElseThrow` on when we remove this duplicated validation

**Pull request closed without merging**